### PR TITLE
Extend Prometheus Support to rest of Platform

### PR DESCRIPTION
**Pull request by nerdynick**
# Description

This extends the jmxexporter/prometheus logic to the rest of the whole platform. The prometheus configs are based of those originally defined in the confluentinc/cp-helm-charts project. The have only been slightly modified to drop the `cp_` from the names in order to match those of kafka_broker and zookeeper.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Launched in dev/test env to verify that all endpoint response with expected output

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: roles/confluent.schema_registry/tasks/main.yml [hunk 1]
```diff
@@ -103,6 +103,15 @@
...
   notify: restart schema-registry
 
+- name: Deploy JMX Exporter Config File
+  copy:
+    src: "schema_registry.yml"
+    dest: "{{schema_registry_jmxexporter_config_path}}"
+    mode: 0640
+    owner: "{{schema_registry_user}}"
+    group: "{{schema_registry_group}}"
+  when: jmxexporter_enabled|bool
```
Comment (domenicbove) [inline roles/confluent.schema_registry/tasks/main.yml:113]:
```suggestion
  when: schema_registry_jmxexporter_enabled |bool
```

File: roles/confluent.ksql/tasks/main.yml [hunk 2]
```diff
@@ -112,6 +112,25 @@
...
+  notify: restart ksql
+
+- name: Deploy JMX Exporter Config File
+  copy:
+    src: "ksql.yml"
+    dest: "{{ksql_jmxexporter_config_path}}"
+    mode: 0640
+    owner: "{{ksql_user}}"
+    group: "{{ksql_group}}"
+  when: jmxexporter_enabled|bool
```
Comment (domenicbove) [inline roles/confluent.ksql/tasks/main.yml:132]:
```suggestion
  when: ksql_jmxexporter_enabled |bool
```

File: roles/confluent.kafka_rest/tasks/main.yml [hunk 3]
```diff
@@ -115,6 +115,15 @@
...
   notify: restart kafka-rest
 
+- name: Deploy JMX Exporter Config File
+  copy:
+    src: "kafka_rest.yml"
+    dest: "{{kafka_rest_jmxexporter_config_path}}"
+    mode: 0640
+    owner: "{{kafka_rest_user}}"
+    group: "{{kafka_rest_group}}"
+  when: jmxexporter_enabled|bool
```
Comment (domenicbove) [inline roles/confluent.kafka_rest/tasks/main.yml:125]:
```suggestion
  when: kafka_rest_jmxexporter_enabled |bool
```

File: roles/confluent.kafka_connect/tasks/main.yml [hunk 4]
```diff
@@ -106,6 +106,15 @@
...
   notify: restart connect distributed
 
+- name: Deploy JMX Exporter Config File
+  copy:
+    src: "kafka_connect.yml"
+    dest: "{{kafka_connect_jmxexporter_config_path}}"
+    mode: 0640
+    owner: "{{kafka_connect_user}}"
+    group: "{{kafka_connect_group}}"
+  when: jmxexporter_enabled|bool
```
Comment (domenicbove) [inline roles/confluent.kafka_connect/tasks/main.yml:116]:
```suggestion
  when: kafka_connect_jmxexporter_enabled |bool
```

File: roles/confluent.schema_registry/tasks/main.yml (hunk 1 above)
Comment (domenicbove) [reply roles/confluent.schema_registry/tasks/main.yml:113]: can you confirm in zk and kafka tasks, they use the proper variables?

File: roles/confluent.ksql/tasks/main.yml [hunk 5]
```diff
@@ -112,6 +112,25 @@
     group: "{{ksql_group}}"
   notify: restart ksql
 
+- name: Ksql Jaas Config
```
Comment (domenicbove) [inline roles/confluent.ksql/tasks/main.yml:115]: please remove the jaas config task, this template is removed/ no longer needed in 55x ksql

**Comment by JumaX** [comment]
As an additional note, we should look at adding Prometheus for the upgrade playbooks as well.  This way we are consistent across the stack, if Prometheus is customer's preferred monitoring agent.

**Pull request merged**